src/keydra/providers/github.py

```diff
@@ -0,0 +1,238 @@
+import copy
+import json
+import yaml
+
+from keydra.clients.github import GithubClient
+
+from keydra.providers.base import BaseProvider
+from keydra.providers.base import exponential_backoff_retry
+
+from keydra.exceptions import ConfigException
+from keydra.exceptions import DistributionException
+from keydra.exceptions import RotationException
+
+from keydra.logging import get_logger
+
+
+LOGGER = get_logger()
+
+ACCT_USERNAME_TAG = 'account_username'
+
+
+class Client(BaseProvider):
+    def __init__(self, session=None, credentials=None, **kwargs):
+        self._client = GithubClient(
+            user=credentials['username'],
+            passwd=credentials['password']
+        )
+
+    def _distribute_repository_secret(self, secret, dest):
+        config = dest['config']
+
+        env_vars = self._client.list_repo_variables(
+            config['repository'],
+            org=config[ACCT_USERNAME_TAG]
+        )
+        existing_variable = None
+
+        for remote_var in env_vars:
+            if remote_var['name'] == dest['key']:
+                existing_variable = remote_var
+                break
+
+        try:
+            if existing_variable is None:
+                LOGGER.info(
+                    '{} not present in Github[{}], adding.'.format(
+                        dest['key'], config['repository']
+                    )
+                )
+
+                if not dest['source'] in secret:
```

Comment:
This check makes sense even if the variable/secret already exists too; otherwise line 66 will throw. I'd move this if block to the beginning of this function.